File: examples/python/best-practices/consistent-naming-patterns.py
# Pattern: verb_noun for operations
context.step(validate_order(order_id), name="validate_order")
context.step(process_payment(amount), name="process_payment")

# Pattern: noun_action for callbacks
context.create_callback(name="payment_callback")
context.create_callback(name="approval_callback")

# Pattern: descriptive_wait for waits
context.wait(Duration.from_seconds(30), name="payment_confirmation_wait")

File: examples/java/best-practices/consistent-naming-patterns.java
// Pattern: verb_noun for operations
ctx.step("validate_order", Map.class, stepCtx -> validateOrder(orderId));
ctx.step("process_payment", Map.class, stepCtx -> processPayment(amount));

// Pattern: noun_action for callbacks
ctx.createCallback("payment_callback", Map.class);
ctx.createCallback("approval_callback", Map.class);

// Pattern: descriptive_wait for waits
ctx.wait("payment_confirmation_wait", Duration.ofSeconds(30));
